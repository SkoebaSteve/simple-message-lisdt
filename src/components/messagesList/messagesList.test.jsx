/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import sinon from 'sinon'
import toJson from 'enzyme-to-json'
import MessageList from './'

const messages = [{
  id: '1',
  type: 'important',
  date: '2018-02-16',
  content: 'this is a very important message',
}, {
  id: '2',
  type: 'less important',
  date: '2018-02-14',
  content: 'this is a less important message',
}]

it('Has the default values.', () => {
  const didMount = sinon.spy()
  // shallow has access to the lifecycle from enzyme 3: https://github.com/airbnb/enzyme/blob/master/docs/api/shallow.md
  const wrapper = shallow(<MessageList messages={messages} getMessages={didMount} />)
  expect(toJson(wrapper)).toMatchSnapshot()
  expect(didMount.callCount).toEqual(1)
})

