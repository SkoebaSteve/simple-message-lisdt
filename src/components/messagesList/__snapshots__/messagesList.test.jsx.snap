// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Has the default values. 1`] = `
<ul>
  <li
    key="1"
  >
    <Message
      content="this is a very important message"
      date="2018-02-16"
      id="1"
      type="important"
    />
  </li>
  <li
    key="2"
  >
    <Message
      content="this is a less important message"
      date="2018-02-14"
      id="2"
      type="less important"
    />
  </li>
</ul>
`;
